src/main/cpp/src/json_parser.cuh

```diff
@@ -1794,21 +1699,12 @@ class json_parser {
   int64_t context_stack;
   int stack_size = 0;
 
+  // TODO remove if possible
   // save current token start pos, used by coping current token text
-  char const* current_token_start_pos;
+  cudf::size_type current_token_start_pos;
```

Comment:
Both the following variables can be removed.
```cpp
cudf::size_type current_token_start_pos;
cudf::size_type number_token_len;
```

If postpone the validatation into `write_raw` and `copy_current_structure`
Then we can remove these two variables.

Current logic is:
Validation a token will go to the end char of the token and save the start pos the the token. If need to copy the current token, have to go back and scan the token again.

Updated logic will be:
Change current  token to expected token by only detect the first char of the next token. Postpone the validatation into `write_raw` and `copy_current_structure`.
e.g.:
{'k' : '123456789'}
When executing `next_token` when current pos is pointing to the string '123456789', just scanning 1st char ' decides the next expected token is String. Let curr pos point to the start of '123456789' and set current expected token as STRING. Then when executing `write_raw` and `copy_current_structure`, do the validation when copying.
After change, the `get_json_object` will only scan the whole JSON 1 time without go back.